### PR TITLE
Rename check_no_unwrap.sh to check_no_panic.sh

### LINKED ISSUE #10930: Rename check_no_unwrap.sh to check_no_panic.sh
The `etc/ci/check_no_unwrap.sh` script checks that we don't call `unwrap()`, but after #10902 will also check for `panic!()`. It should be renamed to `etc/ci/check_no_panic.sh` to reflect the expanded scope.

We'll also need to update the `etc/ci/buildbot_steps.yml` file.

Blocked on servo/saltfs#316, #10902.

### PR DESCRIPTION
**Pull request by 6br**
Hello, I fixed #10930.

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10942)

### CODE REVIEW

File: etc/ci/check_no_panic.sh [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 cd $(git rev-parse --show-toplevel) # cd into repo root so make sure paths works in any case
 
-# files that should not contain "unwrap"
+# files that should not contain "panic"
 FILES=("components/compositing/compositor.rs"
        "components/compositing/pipeline.rs"
        "components/compositing/constellation.rs")
 
-! grep -n "unwrap(\|panic!(" "${FILES[@]}"
+! grep -n "panic(\|panic!(" "${FILES[@]}"
```
Comment (KiChjang) [inline etc/ci/check_no_panic.sh]: This looks very wrong to me.

**Comment by KiChjang** [comment]
Could you also claim the issue saying that you're working on it next time? This is to prevent having two or more people working on the same issue.

**Comment by 6br** [comment]
I'm sorry for not having written in advance, and I'll take care from now on.

File: etc/ci/check_no_panic.sh [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
-# Make sure listed files do not contain "unwrap"
+# Make sure listed files do not contain "panic"
 set -o errexit
 set -o nounset
 set -o pipefail
 
 cd $(git rev-parse --show-toplevel) # cd into repo root so make sure paths works in any case
 
-# files that should not contain "unwrap"
+# files that should not contain "panic"
```
Comment (aneeshusa) [inline etc/ci/check_no_panic.sh]: I actually think we can get rid of this comment, the first one makes this redundant.

File: etc/ci/check_no_panic.sh [hunk 3]
```diff
@@ -1,13 +1,13 @@
 #!/bin/bash
 #
-# Make sure listed files do not contain "unwrap"
+# Make sure listed files do not contain "panic"
```
Comment (aneeshusa) [inline etc/ci/check_no_panic.sh]: Let's say `...do not use unwrap() or panic!()` instead.

**Comment by aneeshusa** [comment]
@6br thanks for the PR! As I mentioned in the issue, this is blocked on servo/saltfs#316, so it may be a while before this is merged. (It's meant to be an example/test run for having CI steps in tree - this isn't a particularly urgent change.)

Also, good catch about updating `.travis.yml`, and good thinking about updating the comment!

**Comment by 6br** [comment]
Thanks for your review. I fixed the comments.

**Comment by emilio** [comment]
Is this still active?

@aneeshusa, it seems to be awaiting for your review (though it now needs a rebase from @6br too).

**Comment by jdm** [comment]
Still blocked on https://github.com/servo/saltfs/issues/316.

**Comment by emilio** [comment]
Huh, true, sorry for not noticing

**Comment by aneeshusa** [comment]
servo/saltfs#316 has been fixed and it looks like all the kinks have been worked out, so this is no longer blocked. Thanks for your patience while that was fixed!

This needs one more change before merging: we need to update the steps for `linux-dev` in `etc/ci/buildbot_steps.yml`.

**Comment by 6br** [comment]
Sure, I fixed `etc/ci/buildbot_steps.yml`.

**Pull request merged**